lib/species.rb

```diff
@@ -0,0 +1,76 @@
+require_relative 'realms'
+class Species
+
+  #Order of keys matters
+  KEYS = %w(ID_NO HEX_ID binomial category).concat(Realms::REALMS).concat(['common_name']).freeze
+  DOWNLOAD_KEYS = {
+    'HEX_ID' => 'Hex ID',
+    'binomial' => 'Species name',
+    'common_name' => 'Common name',
+    'category' => 'IUCN Category',
+    'realms' => 'Realms',
+    'redlist_link' => 'Link to IUCN RedList'
+  }.freeze
+  IUCN_CATEGORIES = {
+    'CR' => 'Critically Endangered',
+    'EN' => 'Endangered',
+    'VU' => 'Vulnerable',
+    'NT' => 'Near Threatened',
+    'DD' => 'Data Deficient'
+  }.freeze
+  THREATENED = %w(CR EN VU).freeze
+
+  # Make KEYS plus :realms accessible instance attributes
+  attr_accessor *KEYS.map(&:downcase).map(&:to_sym) << :realms
```

Comment:
Why did you use the splat operator here?

Reply:
It was necessary as `attr_accessor` doesn't accept an array as parameter